tasks/main.yml

```diff
@@ -58,12 +58,15 @@
   with_dict: riak_conf
 
 - name: Tag riak.conf as ansible managed
-  lineinfile: "dest=/etc/riak/riak.conf regexp='^## Ansible Managed.+$'  line='## Ansible Managed\n' insertbefore='BOF'"
+  lineinfile: dest=/etc/riak/riak.conf regexp='^## Ansible Managed.*$'  line='## Ansible Managed' insertbefore='BOF'
   tags: configfiles
 
-- name: Make certain Riak is set to start on boot
+- name: Make certain Riak is started and set to start on boot
   service: name=riak enabled=yes state=started
 
+- name: Wait for Riak to start up before continuing
+  wait_for: delay=5 timeout=30 host={{ inventory_hostname }} port={{ riak_pb_port }} state=started
```

Comment:
When I was trying out this Riak role against half a dozen virtual machines, all hosted on the same box, I thought I was hitting issues with it a few times. But rarely. And it may have been something else.

Apart from that, I found it was quite handy to spot when Riak *wasn't* starting up, due to earlier errors. As sometimes the "service" command was returning successful, but then Riak wasn't running properly. (For instance, if you provided an invalid node name)